src/components/CreateQuestionView.tsx

```diff
@@ -0,0 +1,88 @@
+import { useAtom } from "jotai"
+import { useState } from "react"
+import { Box, Flex, Group, Modal } from "@mantine/core"
+import { InteractiveQuestion } from "@metabase/embedding-sdk-react"
+
+import { ThemedButton } from "./ThemedButton"
+
+import { isSaveModalOpenAtom } from "../store/save"
+
+export const CreateQuestionView = () => {
+  const [isVisualization, setIsVisualization] = useState(false)
+  const [isSaveModalOpen, setSaveModalOpen] = useAtom(isSaveModalOpenAtom)
+
+  const [isVisualizationButtonShown, setVisualizationButtonShown] =
+    useState(false)
+
+  const onApplyEditorChanges = () => {
+    setIsVisualization(true)
+
+    // Do not show the "show visualization/editor" button until the user has visualized the question.
+    // This prevents users from seeing the "Question not found" indicator.
+    if (!isVisualization) {
+      setVisualizationButtonShown(true)
+    }
+  }
+
+  return (
+    <Box>
+      <Flex
+        w="100%"
+        justify="space-between"
+        pb={14}
+        direction={{ base: "column", sm: "row" }}
+        rowGap="sm"
+      >
+        <Group gap="xs">
+          <InteractiveQuestion.Title />
+        </Group>
+
+        <Group gap="xs">
+          {isVisualizationButtonShown && (
+            <ThemedButton
+              size="compact-sm"
+              onClick={() => setIsVisualization(!isVisualization)}
+            >
+              Show {isVisualization ? "editor" : "visualization"}
+            </ThemedButton>
+          )}
+
+          <ThemedButton
+            size="compact-sm"
+            onClick={() => setSaveModalOpen(true)}
+          >
+            Save
+          </ThemedButton>
+
+          <InteractiveQuestion.ResetButton />
+        </Group>
+      </Flex>
+
+      {isVisualization && (
+        <Box h="500px">
+          <InteractiveQuestion.FilterBar />
+          <InteractiveQuestion.QuestionVisualization />
+        </Box>
+      )}
+
+      {!isVisualization && (
+        <InteractiveQuestion.Editor onApply={onApplyEditorChanges} />
+      )}
+
+      {isSaveModalOpen && (
+        <Modal
+          opened={isSaveModalOpen}
+          onClose={() => setSaveModalOpen(false)}
+          classNames={{ content: "bg-background" }}
+          withCloseButton={false}
+        >
+          <Box bg="background">
+            <InteractiveQuestion.SaveQuestionForm
+              onClose={() => setSaveModalOpen(false)}
```

Comment:
- I just noticed this `onClose` prop is called only when the form is _cancelled_, not when it has succeeded.
- I also noticed that the two buttons does not look equal in size. We should fix that in the SDK as well.
- I wonder if we can have a hook that creates the question based on this question context (that can be used in this `CreateQuestionView` component for instance), so users does not have to use our `SaveQuestionForm` component.

<img src="https://github.com/user-attachments/assets/140b1ac6-0c32-4f00-bd1e-cb7fff20d449" width="300">